File: 3 Longest Substring Without Repeating Characters(No Python yet)/lengthOfLongestSubstring.java
public class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        int l = 0, r = -1;
        int res = 0;
        int[] freq = new int[256]; //Java will initiate the array with 0
        
        while( l < s.length()){ // Note that s is a String object
            if( r+1 < s.length() && freq[s.charAt(r+1)] == 0)
                freq[s.charAt(++r)] ++;
            else
                freq[s.charAt(l++)] --;
            res = Math.max(res, r-l+1); // Note that in Java it is Math.max
        }
        return res;
    }
}

File: 3 Longest Substring Without Repeating Characters(No Python yet)/lengthOfLongestSubstring.cpp
// 8 ms, faster than 94.42%
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
        // Record the frequency of ASCII code character in the sliding window
        int freq[256] = {0};
        int l = 0, r = -1; //The sliding window is s[l...r]
        int res = 0;
        
        while(l < s.size()){
            if( r+1 < s.size() && freq[s[r+1]] == 0 ){
                freq[s[++r]] ++;
            }
            else
                freq[s[l++]] --;
            res = max(res, r-l+1);
        }
        return res;
    }
};